lib/BaseController.js

```diff
@@ -543,8 +543,19 @@ module.exports = class BaseController {
     if (liveResource) {
       let debug = objectPath.get(liveResource, ['metadata', 'labels', 'deploy.razee.io/debug']) ||
         objectPath.get(liveResource, ['metadata', 'labels', 'kapitan.razee.io/debug'], 'false');
+      if (objectPath.get(file, ['metadata', 'annotations']) === null) {
+        objectPath.set(file, ['metadata', 'annotations'], {});
+      }
       if (debug.toLowerCase() === 'true') {
+        this.log.warn(`${uri}: Debug enabled on resource, adding deploy.razee.io/pending-configuration`);
+        await this.patchSelf({ metadata: { annotations: { 'deploy.razee.io/pending-configuration': JSON.stringify(file) } } });
+        await this.modified();
```

Comment:
```suggestion
```
unless you had a specific reason for re-running the cycle, this isn't necessary since the `patchSelf` applies the update to the live resource.